doc/experiment.rst

```diff
@@ -1,327 +1,418 @@
+creation:
+
+1. Manual Model Appending
+     A technique that allows users to create and add model instances to an ensemble, offering a level
+     of customization in ensemble design.
+2. Parameter Expansion
+     A technique that allows users to set parameter values and control how the parameter values
+     spread across the ensemble members by specifying a permutation strategy.
+3. The Utilization of Replicas
+     A technique that allows users to create identical or closely related models within an ensemble. Users can assess
+     how a system responds to the same set of parameters under multiple instances.
+
+.. note::
+  For more information and instruction on ensemble creation methods, navigate to the Ensemble documentation page.
+
+==================
+Experiment Example
+==================
+.. compound::
+  In the following subsections, we provide an example of using SmartSim to automate the
+  deployment of an HPC workload consisting of a ``Model`` and standard ``Orchestrator``.
+
+  Continue to the example to:
+
+  .. list-table:: Experiment example contents
+   :widths: auto
+   :header-rows: 1
+
+   * - Initialize
+     - Start
+     - Stop
+   * - a workflow (``Experiment``)
+     - the in-memory database (``Orchestrator``)
+     - the in-memory database (``Orchestrator``)
+   * - a in-memory database (``Orchestrator``)
+     - the workload (``Model``)
+     - 
+   * - a workload (``Model``)
+     - 
+     - 
+
+Initialize
+----------
+.. compound::
+  To create a workflow, we *initialize* an ``Experiment`` object
+  once at the beginning of the Python driver script.
+  To create an Experiment, we specify a name
+  and the system launcher with which all entities will be executed.
+  Here, we are running the example on a Slurm machine and as such will
+  set the `launcher` argument to `slurm`.
+
+  .. code-block:: python
+
+      from smartsim import Experiment
+      from smartsim.log import get_logger
+
+      # Initialize an Experiment
+      exp = Experiment("name-of-experiment", launcher="slurm")
+      # Initialize a SmartSim logger
+      smartsim_logger = get_logger("tutorial-experiment")
+
+  We also initialize a SmartSim logger. We will use the logger throughout the experiment
+  to monitor the entities.
+
+.. compound::
+  Next, we will launch a SmartSim in-memory database called an ``Orchestrator``.
+  To *initialize* an ``Orchestrator`` object, use the ``Experiment.create_database()``
+  function. We will create a single-sharded database and therefore will set
+  the argument `db_nodes` to 1. SmartSim will assign a `port` to the database
+  and attempt to detect your machine's interface if values are not provided to the ``Experiment.create_database()`` factory method.
+
+  .. code-block:: python
+
+      # Initialize an Orchestrator
+      database = exp.create_database(db_nodes=1)
+      # Create an output directory
+      exp.generate(database)
```

Comment:
I think we may want to put the ``exp.generate()`` call right before exp.start because we want the database and the model to be genreated.  I think that's the preferred flow.